### PR TITLE
Osc modularizing

### PR DESCRIPTION
**Pull request by studywolf**
Breaking up the OSC code into some smaller functions so that tests can more easily be written.

- fixed several errors in the code around velocity limiting with orientation and also the integrated error term
- vmax is now calculated using the norm of the xyz and abg signals, not individually across each dimension
- also now vmax is a list of two terms, the max velocity and max angular velocity 
- fixed the jaco2 singularity issue we were running into
- fixed default value of `encoders=nengo.Default` in `DynamicsAdaptation` instead of `None` to work with current versions of Nengo
- a bunch of style cleanup

### CODE REVIEW

**Comment by studywolf** [comment]
Because this has a few fixes for the main osc code, as well as speed improvements, I would like to get this PR merged in to master and have adding the tests in a future PR

File: abr_control/controllers/osc.py [hunk 1]
```diff
@@ -9,44 +9,42 @@ class OSC(Controller):
...
-    integrated_error : float list, optional (Default: None)
+    integrated_error: float list, optional (Default: None)
         task-space integrated error term
     """
-    def __init__(self, robot_config, kp=1, ko=None, kv=None, ki=0, vmax=None,
-                 ctrlr_dof=None, null_controllers=None,
-                 use_g=True, use_C=False):
+    def __init__(self, robot_config, kp=1, ko=None, kv=None, ki=0,
+                 vmax=None, ctrlr_dof=None, null_controllers=None,
+                 use_g=True, use_C=False, orientation_algorithm=0):
```
Comment (p3jawors) [inline abr_control/controllers/osc.py]: missing orientation_algorithm in docstring

File: abr_control/controllers/osc.py [hunk 2]
```diff
@@ -56,164 +54,214 @@ def __init__(self, robot_config, kp=1, ko=None, kv=None, ki=0, vmax=None,
...
-            # # get rotation matrix for the target orientation
-            # R_d = transformations.euler_matrix(
-            #     target[3], target[4], target[5], axes='rxyz')[:3, :3]
-            # R_ed = np.dot(R_e.T, R_d)  # eq 24
-            # q_ed = transformations.unit_vector(
-            #     transformations.quaternion_from_matrix(R_ed))
-            # u_task[3:] = -self.ko / self.kp * np.dot(R_e, q_ed[1:])  # eq 34
+        # task space integrated error term
+        if self.ki != 0:
+            self.integrated_error -= u_task
```
Comment (p3jawors) [inline abr_control/controllers/osc.py]: why do we subtract u_task from the integrated error?

Comment (studywolf) [reply abr_control/controllers/osc.py]: fixed up signs

**Comment by p3jawors** [comment]
LGTM

**Pull request merged**